src/foundry/common/prosemirror/content-link-plugin.d.mts

```diff
@@ -42,3 +34,14 @@ declare class ProseMirrorContentLinkPlugin extends ProseMirrorPlugin {
    */
   protected _onDrop(view: EditorView, event: DragEvent, slice: Slice, moved: boolean): void | true;
 }
+
+declare namespace ProseMirrorContentLinkPlugin {
+  interface ContentLinkOptions {
+    /** The parent document housing this editor. */
+    document?: foundry.abstract.Document<any, any, any>;
+    /** @defaultValue `false` */
+    relativeLinks?: boolean;
+  }
```

Comment:
I assume explicit undefined is fine, right?

Therefore I would recommend `| undefined` for both of these properties.

Reply:
I'm not sure.  Relativelinks is ok with undefined.  For document, I guess I'd say no... it gets used as:

```    
Object.defineProperty(this, "document", {value: document, writable: false});
```

And `defineProperty` has value as `value?: any`

Reply:
Well is `document` never `undefined`? You can ignore the typing of `defineProperty` here, it probably doesn't account for EEOPT like we do, ironically enough.